src/utils/assertLocationUtils.tsx

```diff
@@ -70,26 +71,45 @@ export const assertLocationTxn = async (
   return finalTxn
 }
 
-export const loadLocationFeeData = async (
+export const loadLocationFeeData = async ({
   nonce = 0,
   accountIntegerBalance = 0,
-  onboardingRecord?: OnboardingRecord,
-) => {
-  const isFree = await hasFreeLocationAssert(nonce, onboardingRecord)
+  onboardingRecord,
+  dataOnly = false,
+}: {
+  nonce?: number
+  accountIntegerBalance?: number
+  onboardingRecord?: OnboardingRecord
+  dataOnly?: boolean
+}) => {
+  let isFree = false
+  if (!dataOnly) {
+    isFree = await hasFreeLocationAssert(nonce, onboardingRecord)
```

Comment:
Should we move this `dataOnly` check to `hasFreeLocationAssert` maybe? I see in the `assertLocationTxn` line 36 is also checking this, does it need a check for dataOnly there too? Maybe in `hasFreeLocationAssert` we return false if its dataOnly?

Reply:
My assumption is that `dataOnly` hotspots won't ever have a free assert. Is that right @cokes518 ?

Reply:
correct. no free asserts.